server/data/restClient.ts

```diff
@@ -165,4 +111,40 @@ export default class RestClient {
         })
     })
   }
+
+  private async postOrPut(
```

Comment:
might be easier to follow if this refactoring were in a separate commit, before the change in behaviour was introduced?

Reply:
That's a good shout 👍